publication-rest/src/test/java/no/unit/nva/publication/create/CreatePublicationFromImportCandidateHandlerTest.java

```diff
@@ -0,0 +1,245 @@
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import no.unit.nva.api.PublicationResponse;
+import no.unit.nva.identifiers.SortableIdentifier;
+import no.unit.nva.model.AdditionalIdentifier;
+import no.unit.nva.model.Contributor;
+import no.unit.nva.model.EntityDescription;
+import no.unit.nva.model.Identity;
+import no.unit.nva.model.Organization;
+import no.unit.nva.model.PublicationDate;
+import no.unit.nva.model.PublicationStatus;
+import no.unit.nva.model.ResearchProject;
+import no.unit.nva.model.ResourceOwner;
+import no.unit.nva.model.Username;
+import no.unit.nva.model.funding.FundingBuilder;
+import no.unit.nva.model.role.Role;
+import no.unit.nva.model.role.RoleType;
+import no.unit.nva.publication.exception.TransactionFailedException;
+import no.unit.nva.publication.model.business.ImportCandidate;
+import no.unit.nva.publication.model.business.ImportStatus;
+import no.unit.nva.publication.model.business.UserInstance;
+import no.unit.nva.publication.service.ResourcesLocalTest;
+import no.unit.nva.publication.service.impl.ResourceService;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import nva.commons.apigateway.AccessRight;
+import nva.commons.apigateway.GatewayResponse;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.NotFoundException;
+import nva.commons.core.Environment;
+import org.apache.hc.core5.http.ContentType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.zalando.problem.Problem;
+
+public class CreatePublicationFromImportCandidateHandlerTest extends ResourcesLocalTest {
+
+    public static final String PUBLICATIONS_TABLE = new Environment().readEnv("TABLE_NAME");
+    private ByteArrayOutputStream output;
+    private Context context;
+    private ResourceService importCandidateService;
+    private ResourceService publicationService;
+    private CreatePublicationFromImportCandidateHandler handler;
+
+    @BeforeEach
+    public void setUp() {
+        super.init(IMPORT_CANDIDATES_TABLE, PUBLICATIONS_TABLE);
+        Environment environment = mock(Environment.class);
+        when(environment.readEnv(ALLOWED_ORIGIN_ENV)).thenReturn("*");
+        importCandidateService = new ResourceService(client, IMPORT_CANDIDATES_TABLE);
+        publicationService = new ResourceService(client, Clock.systemDefaultZone(), SortableIdentifier::next);
```

Comment:
`ResourceService` is "generic" and you can supply a table name, which make me think that ResourceService is a generic interface with create, update, fetch & delete. 
however  getPublicationByIdentifier & getImportCandidateByIdentifier are very specific, and I guess are doing very specific work?

Reply:
You are right, we "reuse" resourceService to make operation on different table, I see no reason to implement new ResourceService that will operate on import-candidate table only.

Reply:
Yeah, I am wondering this myself. Is it because you loose the Import-status attribute from the object so that a method ImportPublication fromResource(Resource resource) method does not work?

Reply:
Then why have specific calls to specific tables inside the "generic" service?
A generic service should then just be doing simple IO from and to storage  (`create`, `update`, `fetch `& `delete`.)

Reply:
Me and Kir are in a discussion of removing everything related to import-candidates to a different repository. 
For now we just have to live with this. 

If we see loads more of "we need to make special implementation in resourceService", I think it's time to pack "import-candidates bags", and move it to a seperate github repo. (we need to make PO prioritize a non-functional task).

@StigNorland (TLDR at the end) The reason for why import-candidates are in publication-api repo.
It was because previously the workflow was just two lambdas, sitting in it's own repo, sending thousands of events to publication-api with requests for publication-api to push publications into the resource table. It worked exactly as great as the event flow does everwhere else in NVA. A couple of months ago we refactored to make import-candidates push to it's own table, as NVA don't need to know about them before they get status imported.
TLDR: Historical stupid reasons.